modules/react/switch/lib/Switch.tsx

```diff
@@ -42,123 +34,142 @@ export interface SwitchProps extends Themeable {
+    },
+  },
+  modifiers: {
+    error: {
+      error: {
+        '& ~ div:first-of-type': {
           boxShadow: `
-            0 0 0 2px ${colors.frenchVanilla100},
-            0 0 0 4px ${errorColors.inner},
-            0 0 0 5px ${errorColors.outer}`,
+              0 0 0 ${px2rem(2)} ${base.frenchVanilla100},
+              0 0 0 ${system.space.x1} ${brand.error.base},
+              0 0 0 ${px2rem(5)} transparent`,
         },
-      }),
-    };
-  }
-);
+      },
+      alert: {
+        '& ~ div:first-of-type': {
+          boxShadow: `
+          0 0 0 ${px2rem(2)} ${base.frenchVanilla100},
+          0 0 0 ${system.space.x1} ${brand.alert.base},
+          0 0 0 ${px2rem(5)} ${brand.alert.darkest}`,
+        },
+      },
+    },
+  },
+});
+
+const SwitchInput = createComponent('input')<SwitchProps>({
+  displayName: 'SwitchInput',
+  Component: ({error, ...elemProps}, ref, Element) => {
+    return <Element ref={ref} {...mergeStyles(elemProps, switchInputStencil({error}))} />;
+  },
+});
 
-const SwitchBackground = styled('div')<Pick<SwitchProps, 'checked' | 'disabled'>>(
-  {
+const switchBackgroundStencil = createStencil({
+  base: {
     boxSizing: 'border-box',
     position: 'absolute',
     display: 'flex',
     alignItems: 'center',
     pointerEvents: 'none',
-    marginTop: space.xxxs,
-    width: switchWidth,
-    height: switchHeight,
-    borderRadius: borderRadius.circle,
-    padding: '0px 2px',
+    marginTop: system.space.x1,
+    width: system.space.x8,
+    height: system.space.x4,
+    borderRadius: system.shape.round,
+    padding: `${system.space.zero} ${px2rem(2)}`,
     transition: 'background-color 200ms ease',
+    backgroundColor: base.licorice200,
   },
-  ({
-    checked,
-    disabled,
-    theme: {
-      canvas: {
-        palette: {primary: themePrimary},
-      },
-    },
-  }) => {
-    if (checked) {
-      return {
-        backgroundColor: disabled ? themePrimary.light : themePrimary.main,
-      };
-    } else {
-      return {
-        backgroundColor: disabled ? colors.soap400 : colors.licorice200,
-      };
-    }
-  }
-);
+});
 
-const SwitchCircle = styled('div')<Pick<SwitchProps, 'checked' | 'theme'>>(
-  {
-    width: circleSize,
-    height: circleSize,
-    borderRadius: borderRadius.circle,
-    ...depth[1],
+const SwitchBackground = createComponent('div')({
+  displayName: 'SwitchBackground',
+  Component: ({children, ...elemProps}, ref, Element) => {
+    return (
+      <Element ref={ref} {...mergeStyles(elemProps, switchBackgroundStencil())}>
+        {children}
+      </Element>
+    );
+  },
+});
+
+const switchCircleStencil = createStencil({
+  base: {
+    width: system.space.x3,
+    height: system.space.x3,
+    borderRadius: system.shape.round,
+    boxShadow: system.depth[1],
     transition: 'transform 150ms ease',
     pointerEvents: 'none',
+    backgroundColor: brand.primary.accent,
+    transform: 'translateX(0)',
```

Comment:
Shouldn't this also uses token `translateX(${system.space.zero})` for consistency? Not sure, so your decision 🤔

Reply:
It should for consistency you're right.